src/Internal.AspNetCore.BuildTools.Tasks/Run.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Build.Framework;
+
+namespace Microsoft.AspNetCore.BuildTools
+{
+    /// <summary>
+    /// A better version of the Exec task. See <seealso cref="RunBase" /> for more arguments.
```

Comment:
@pranavkm I add a few things here for common gotchas of Exec.

1. Argument escaping. Exec.Command is a string, which means callers have to manually `&quot;` escape args. Run.Arguments will take care of this, and it will do it right. Wrapping in `&quot;` is not always enough to ensure arguments are escaped. For example, an arg on Windows that ends in `\`. We generally workaround by adding a space between `\` and the quote, but that's not an obivous solution to someone new to MSBuild.
2. MaxRetries. Use case: `<RunDotNet Command="nuget push package.nupkg" MaxRetries="5" />`.
3. Running dotnet commands. `RunDotNet` is shorter syntax, a combination of GetDotNetHost + Exec.